gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -678,12 +680,24 @@ private void checkPartitionCompatibility(Partition desiredTargetPartition, Parti
     List<CopyableFile.Builder> builders = Lists.newArrayList();
     List<SourceAndDestination> dataFiles = Lists.newArrayList();
 
+    Configuration hadoopConfiguration = new Configuration();
+    FileSystem actualSourceFs = null;
+    String referenceScheme = null;
+    String referenceAuthority = null;
+
     for (FileStatus status : paths) {
       dataFiles.add(new SourceAndDestination(status, getTargetPath(status.getPath(), this.targetFs, partition, true)));
     }
 
     for (SourceAndDestination sourceAndDestination : dataFiles) {
-      FileSystem actualSourceFs = sourceAndDestination.getSource().getPath().getFileSystem(new Configuration());
+
+      URI uri = sourceAndDestination.getSource().getPath().toUri();
+      if (actualSourceFs == null || !StringUtils.equals(referenceScheme, uri.getScheme())
```

Comment:
I think that's overkill. Honestly, most of the time there is a single `FileSystem` needed. Just putting this logic in case there is some super-edge case.